src/Base.js

```diff
@@ -0,0 +1,39 @@
+'use strict'
+
+class Base {
+  constructor (name, code, implementation, alphabet) {
+    this.name = name
+    this.code = code
+    this.alphabet = alphabet
+    if (implementation && alphabet) {
+      this.engine = implementation(alphabet)
+    }
+  }
+
+  getName () {
+    return this.name
+  }
+
+  getCode () {
+    return this.code
+  }
+
+  getAlphabet () {
+    return this.alphabet
+  }
```

Comment:
No need for this methods, as they are sync and only return properties that are defined.